Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackIntegrationDriver.swift

```diff
@@ -39,59 +35,131 @@ class SentryUserFeedbackIntegrationDriver: NSObject, SentryUserFeedbackWidgetDel
+        }
+        widget?.setWidget(visible: true, animated: configuration.animations)
     }
-    
-    /**
-     * Removes the feedback widget from the view hierarchy. Useful for cleanup when the widget is no longer needed.
-     */
-    func removeWidget() {
-        
+}
+
+// MARK: SentryUserFeedbackWidgetDelegate
+@available(iOSApplicationExtension 13.0, *)
+extension SentryUserFeedbackIntegrationDriver: SentryUserFeedbackWidgetDelegate {
+    func showForm(presenter: UIViewController) {
+        showForm(presenter: presenter, screenshot: nil)
     }
-    
-    private func validate(_ config: SentryUserFeedbackWidgetConfiguration) {
+}
+
+// MARK: UIAdaptivePresentationControllerDelegate
+@available(iOSApplicationExtension 13.0, *)
+extension SentryUserFeedbackIntegrationDriver: UIAdaptivePresentationControllerDelegate {
+    func presentationControllerDidDismiss(_ presentationController: UIPresentationController) {
+        widget?.setWidget(visible: true, animated: configuration.animations)
+        configuration.onFormClose?()
+    }
+}
+
+// MARK: Private
+@available(iOSApplicationExtension 13.0, *)
+private extension SentryUserFeedbackIntegrationDriver {
+    func showForm(presenter: UIViewController, screenshot: UIImage?) {
+        let form = SentryUserFeedbackFormController(config: configuration, delegate: self, screenshot: screenshot)
+        form.presentationController?.delegate = self
+        widget?.setWidget(visible: false, animated: configuration.animations)
+        presenter.present(form, animated: configuration.animations) {
+            self.configuration.onFormOpen?()
+        }
+    }
+
+    func validate(_ config: SentryUserFeedbackWidgetConfiguration) {
         let noOpposingHorizontals = config.location.contains(.trailing) && !config.location.contains(.leading)
-            || !config.location.contains(.trailing) && config.location.contains(.leading)
+        || !config.location.contains(.trailing) && config.location.contains(.leading)
         let noOpposingVerticals = config.location.contains(.top) && !config.location.contains(.bottom)
-            || !config.location.contains(.top) && config.location.contains(.bottom)
+        || !config.location.contains(.top) && config.location.contains(.bottom)
         let atLeastOneLocation = config.location.contains(.trailing)
-            || config.location.contains(.leading)
-            || config.location.contains(.top)
-            || config.location.contains(.bottom)
+        || config.location.contains(.leading)
+        || config.location.contains(.top)
+        || config.location.contains(.bottom)
         let notAll = !config.location.contains(.all)
         let valid = noOpposingVerticals && noOpposingHorizontals && atLeastOneLocation && notAll
-        #if DEBUG
+#if DEBUG
         assert(valid, "Invalid widget location specified: \(config.location). Must specify either one edge or one corner of the screen rect to place the widget.")
```

Comment:
`m`: Why is this wrapped in a `DEBUG` check?

Reply:
to try to avoid crashing in production. we have various macros like this in objc code, but it's clumsier in swift